go/Dockerfile.build

```diff
@@ -0,0 +1,10 @@
+FROM golang:1.8-alpine
+RUN apk update && apk add --no-cache make git
+ENV GOPATH=/go
+ENV PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+# The project sources
+VOLUME ["/go/src/github.com/docker/hyperkit/go"]
+WORKDIR /go/src/github.com/docker/hyperkit/go
+
+ENTRYPOINT ["make"]
```

Comment:
I tend to cross compile Go code and GOOS and GOARCH are set to OS X in the Makefile. Compiling in a container this way frees me from having to stick all my repositories into GOPATH